src/charm_state.py

```diff
@@ -85,7 +91,26 @@ def from_charm(cls, charm: CharmBase) -> "CharmConfig":
             logger.exception("Invalid runner-storage configuration")
             raise CharmConfigInvalidError("Invalid runner-storage configuration") from err
 
-        return cls(runner_storage=runner_storage)
+        openstack_clouds_yaml_str = charm.config.get(OPENSTACK_CLOUDS_YAML_CONFIG_NAME)
+        if openstack_clouds_yaml_str:
+            try:
+                openstack_clouds_yaml = yaml.safe_load(openstack_clouds_yaml_str)
+            except yaml.YAMLError as exc:
+                logger.error("Invalid openstack-clouds-yaml config: %s.", exc)
+                raise CharmConfigInvalidError(
+                    "Invalid openstack-clouds-yaml config. Invalid yaml."
+                ) from exc
+            try:
+                openstack_manager.initialize(openstack_clouds_yaml)
```

Comment:
The type hint for `initialise` is that `openstack_clouds_yaml` is a `dict` which is not guaranteed at this point - it could be a `list`, `str` or `int` etc. Perhaps checking it is a `dict` is worthwhile here as well? Also potentially means that additional tests could be written here

Reply:
Added tests to check parsed type, thank you!